utils/rowgroup/rowaggregation.cpp

```diff
@@ -687,9 +687,25 @@ void RowAggregation::initialize()
   makeAggFieldsNull(fRow);
 
   // Keep a copy of the null row to initialize new map entries.
-  fRowGroupOut->initRow(&fNullRow, true);
-  fNullRowData.reset(new uint8_t[fNullRow.getSize()]);
-  fNullRow.setData(rowgroup::Row::Pointer(fNullRowData.get()));
+  // MCOL-5429 Use stringstore if the datatype of the groupconcat
+  // field is a long string.
+  if (hasGroupConcat && fRowGroupOut->hasLongString())
+  {
+    fNullRowGroup = *fRowGroupOut;
```

Comment:
I didn't get it. Do you add fNullRowGroup to just use it for init here? Why then do you need to have the fNullRowGroup as RowAggregation attribute?

Reply:
IMHO It is worth to limit the span of this RG and remove it from the list of RowAggregation attributes.

Reply:
`fNullRowData` which is also just used in the else block is also an attribute of RowAggregation. We can keep this as is to maintain the code semantics here.

Reply:
Feel free to remove fNullRowData also b/c these two are wastes RAM in runtime.

Reply:
No. This is not wrecked pattern. The reason fNullRowData is an attribute of the class is because it owns a memory buffer to which fNullRow points. If we remove fNullRowData as a class member and make it local variable with the scope of the else block, then the RGData dtor would get called and destroy the underlying memory buffer, causing fNullRow to point to an invalid memory address.

Reply:
Makes sense. You don't need RG to stay around as RowStorage attribute though, right?

Reply:
`fRowGroup` is tightly coupled with `fRowRGData`, so I am keeping the declarations of both at the same place.

Reply:
I didn't get your logic behind this. IMHO this approach values the form more then the semantics but I won't push on this anymore.

Reply:
If I make `fNullRowGroup` local to the scope of the if block, now `fNullRowGroup.initRow(&fNullRow);` would be invalid. This is because `initRow()` is making some `Row` pointer attributes point to the `RowGroup` objects (`Row::oldOffsets`, `Row::stOffsets` are pointers to their corresponding objects defined in the `RowGroup` class). As soon as `fNullRowGroup` goes out of scope, these `offset` objects are destroyed and the corresponding `Row` pointers to these offsets would now be invalid.

Below is the `initRow` definition. If you look at the if-else blocks, you will see the pointer assignments to the offset attributes:
```c++
void RowGroup::initRow(Row* r, bool forceInlineData) const                                                                                                                                 
{                                                                                                                                                                  
  r->columnCount = columnCount;                                                                                                                                    
                                                                                                                                                                                   
  if (LIKELY(!types.empty()))                                                                                                                                                                                                                       
  {                                                                                                                                                                                                                                      
    r->colWidths = (uint32_t*)&colWidths[0];                                                                                                                                                                                       
    r->types = (execplan::CalpontSystemCatalog::ColDataType*)&(types[0]);                                                                                                
    r->charsetNumbers = (uint32_t*)&(charsetNumbers[0]);                                                                                                                                                                                            
    r->charsets = (CHARSET_INFO**)&(charsets[0]);                                                                                                                         
    r->scale = (uint32_t*)&(scale[0]);                                                                                                                                                                                                              
    r->precision = (uint32_t*)&(precision[0]);                                                                                                                      
  }                                                                                                                                                                
                                                                                                                                                                                                     
  if (forceInlineData)                                                                                                                                              
  {                                                                                                                                                                                                                                                 
    r->useStringTable = false;                                                                                                                                                                                       
    r->oldOffsets = (uint32_t*)&(oldOffsets[0]);                                                                                                                   
    r->stOffsets = (uint32_t*)&(stOffsets[0]);                                                                                                                                                
    r->offsets = (uint32_t*)&(oldOffsets[0]);                                                                                                                                                           
  }                                                                                                                                                                
  else                                                                                                                                                                                     
  {                                                                                                                                                                
    r->useStringTable = useStringTable;                                                                                                                                            
    r->oldOffsets = (uint32_t*)&(oldOffsets[0]);                                                                                                                                                                                                    
    r->stOffsets = (uint32_t*)&(stOffsets[0]);                                                                                                                                                                                           
    r->offsets = offsets;                                                                                                                                                                                                          
  }                                                                                                                                                                      
                                                                                                                                                                                                                   
  r->hasLongStringField = hasLongStringField;                                                                                                                             
  r->sTableThreshold = sTableThreshold;                                                                                                                             
  r->forceInline = forceInline;                                                                                                                                    
  r->hasCollation = hasCollation;                                                                                                                                                                                            
}
```

Reply:
This RowGroup placement makes sense to me now. Thank you for the patience and the detailed explanation.